pyccel/codegen/wrapper/fortran_to_c_wrapper.py

```diff
@@ -172,7 +172,8 @@ def _wrap_Module(self, expr):
         self.scope = mod_scope
 
         # Wrap contents
-        funcs_to_wrap = expr.funcs
+        funcs_to_wrap = [f for f in expr.funcs if f.is_annotated]
```

Comment:
this is needed when we use the `inline` decorator without calling the function, so we just ignore the function and don't annotate it

Reply:
I'm not sure you understood my point. If I translate:
```python
@inline
def f(a : int | float):
    print(a)

if __name__ == '__main__':
     f(3)
```
then you will only annotate `def f(a : int)` however in the wrapper you then wrap `def f(a : int)` so once this file is translated I can do:
```python
from inlined import f
f(4)
```
But not:
```python
from inlined import f
f(4.0)
```
This seems quite counter-intuitive to me. I would expect that either inline functions shouldn't be wrapped or that all versions of the inline function should be wrapped.

But unless all are wrapped this is a change in behaviour, so it must be mentioned in the CHANGELOG.
Regardless the behaviour should also be documented in docs/decorators.md

Reply:
ah no, we never wrap inline functions and they can not be imported because we don't print them,  so they are never exposed to the user directly, the code replaces the function call with the function body, that is why we check if the function is annotated  i.e is_inline in this case

Reply:
> ah no, we never wrap inline functions

They are no longer wrapped from C to Python, but here you are still wrapping them from Fortran to C.

Reply:
no we also ignore them

Reply:
Can you point me at the line where they are filtered out please. It is not this line right?

Reply:
Never mind you have updated this line. Now I agree

Reply:
The CHANGELOG has already been updated, but for the user, there will be no change, it is an internal thing, and they will see the same results as before, it is just faster

Reply:
> for the user, there will be no change

This is not true. In the devel branch the following works:
```
$ printf "@inline\ndef f():\n    print('hello')" > tmp_inline.py
$ pyccel tmp_inline.py
$ python3 -c "from tmp_inline import f; f()"
hello
```
In this branch it raises an `ImportError`:
```
$ printf "@inline\ndef f():\n    print('hello')" > tmp_inline.py
$ pyccel tmp_inline.py
$ python3 -c "from tmp_inline import f; f()"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: cannot import name 'f' from 'tmp_inline' (/home/emily/Code/test/tmp/tmp_inline.cpython-310-x86_64-linux-gnu.so)
```

This is definitely a change for the user which is why I would like it to be documented

Reply:
how is that possible? we have always returned an empty string in the `_print_FunctionDef` of the `fcode`

Reply:
The wrapper inlines the code into the wrapper function (which is not an inline function)

Reply:
Great thanks :) . Don't forget the CHANGELOG

Reply:
[944672c](https://github.com/pyccel/pyccel/pull/1728/commits/944672c71d3f088fef373685c025de8a61b19f6a)